### PR TITLE
tstest/integration/vms: make first end to end test

### PR DESCRIPTION
**Pull request by Xe**
This makes sure `tailscale status` and `tailscale ping` works. It also
switches goexpect to use a batch instead of manually banging out each
line, which makes the tests so much easier to read.

Signed-off-by: Christine Dodrill <xe@tailscale.com>

### CODE REVIEW

**Comment by Xe** [comment]
Seems to be oddly flaky. Will investigate and fix by tomorrow. I wonder if it's me assuming things wrongly about how the shell config in each distro works.

**Comment by Xe** [comment]
AHA! I found the reason for the flakiness. I assumed that the way the expect library worked would that it would be line by line, like TCL expect. However sometimes the shell prompt gets drawn quick enough that it's in the previous testing batch. It actually works fine!

File: tstest/integration/vms/vms_test.go [hunk 1]
```diff
@@ -729,35 +731,48 @@ func testDistro(t *testing.T, loginServer string, d Distro, signer ssh.Signer, i
...
-	err = e.Send(fmt.Sprintf("tailscale up --login-server %s\n", loginServer))
-	if err != nil {
-		t.Fatalf("%d: can't send tailscale up command: %v", port, err)
-	}
-	_, _, err = e.Expect(regexp.MustCompile(`Success.`), timeout)
+		batch = append(batch, &expect.BSnd{S: "systemctl start tailscaled.service\n"})
+	}
+
+	batch = append(batch,
+		&expect.BExp{R: `(\#)`},
```
Comment (crawshaw) [inline tstest/integration/vms/vms_test.go:752]: You can remove some boilerplate here with:

```
batch = append(batch, []*expect.BExp{
        {R: `(\#)`},
        {S: `...`},
}...)
```

File: tstest/integration/vms/vms_test.go [hunk 2]
```diff
@@ -874,6 +889,57 @@ func TestDeriveBindhost(t *testing.T) {
...
+	cmd.Env = append(os.Environ(), "NOTIFY_SOCKET="+filepath.Join(dir, "notify_socket"))
+
+	err := cmd.Start()
+	if err != nil {
+		t.Fatalf("can't start tailscaled: %v", err)
+	}
+
+	t.Cleanup(func() {
+		cmd.Process.Kill()
+		cmd.Process.Kill()
```
Comment (crawshaw) [inline tstest/integration/vms/vms_test.go:911]: Why is this here twice?

File: tstest/integration/vms/vms_test.go (hunk 1 above)
Comment (Xe) [reply tstest/integration/vms/vms_test.go:752]: They are different types for sending and receiving.

Comment (crawshaw) [reply tstest/integration/vms/vms_test.go:752]: Oh so they are. I misread it as using different fields on the same type.

File: tstest/integration/vms/vms_test.go (hunk 2 above)
Comment (Xe) [reply tstest/integration/vms/vms_test.go:911]: Overly defensive programming! I'm gonna prune it.

File: tstest/integration/vms/vms_test.go [hunk 3]
```diff
@@ -874,6 +889,56 @@ func TestDeriveBindhost(t *testing.T) {
...
+	)
+
+	cmd.Env = append(os.Environ(), "NOTIFY_SOCKET="+filepath.Join(dir, "notify_socket"))
+
+	err := cmd.Start()
+	if err != nil {
+		t.Fatalf("can't start tailscaled: %v", err)
+	}
+
+	t.Cleanup(func() {
```
Comment (crawshaw) [inline tstest/integration/vms/vms_test.go:909]: nit: `t.Cleanup(cmd.Process.Kill)`

**Review by crawshaw — APPROVED** [review]

File: tstest/integration/vms/vms_test.go (hunk 3 above)
Comment (crawshaw) [reply tstest/integration/vms/vms_test.go:909]: oh nevermind, Kill probably returns an error.

Comment (Xe) [reply tstest/integration/vms/vms_test.go:909]: I can't do that, Kill returns an error because the kernel can deny a kill for permissions reasons. In this case I am assuming that this will work because this is a child process and the tests aren't running with the kernel capability needed to change user ids.
